adapter_docs/adapter_composition.md

```diff
@@ -0,0 +1,145 @@
+```
+
+The basic building blocks of the more advanced setups are simple objects derived from `AdapterCompositionBlock`,
+each representing a different possibility to combine single adapters.
+They are presented in more detail in the following.
+
+## `Stack`
+
+The `Stack` block can be used to stack multiple adapters on top of each other.
+This kind of adapter composition is used e.g. in the _MAD-X_ framework for cross-lingual transfer [(Pfeiffer et al., 2020)](https://arxiv.org/pdf/2005.00052.pdf), where language and task adapters are stacked on top of each other.
+For more, check out [this Colab notebook](https://colab.research.google.com/github/Adapter-Hub/adapter-transformers/blob/master/notebooks/04_Cross_Lingual_Transfer.ipynb) on cross-lingual transfer.
+
+In the following example, we stack the adapters `a`, `b` and `c` so that in each layer, the input is first passed through `a`, the output of `a` is then inputted to `b` and the output of `b` is finally inputted to `c`.
+
+```python
+import transformers.adapters.composition as ac
+
+// ...
+
+model.add_adapter("a")
+model.add_adapter("b")
+model.add_adapter("c")
+
+model.active_adapters = ac.Stack("a", "b", "c")
+```
+
+In v1.x of `adapter-transformers`, stacking adapters was done using a list of adapter names, i.e. the example from above would be defined as `["a", "b", "c"]`.
+For backwards compatibility, you can still do this, although it is recommended to use the new syntax.
+
+## `Fuse`
+
+The `Fuse` block can be used to activate a fusion layer of adapters.
+_AdapterFusion_ is a non-destructive way to combine the knowledge of multiple pre-trained adapters on a new downstream task, proposed by [Pfeiffer et al., 2021](https://arxiv.org/pdf/2005.00247.pdf).
+In the following example, we activate the adapters `d`, `e` and `f` as well as the fusion layer that combines the outputs of all three.
+The fusion layer is added 
+
+```python
+import transformers.adapters.composition as ac
+
+// ...
+
+model.add_adapter("d")
+model.add_adapter("e")
+model.add_adapter("f")
+model.add_fusion(["d", "e", "f"])
+
+model.active_adapters = ac.Fuse("d", "e", "f")
+```
+
+```eval_rst
+.. important::
+    Fusing adapters with the ``Fuse`` block only works successfully if an adapter fusion layer combining all of the adapters listed in the ``Fuse`` has been added to the model.
+    This can be done either using ``add_fusion()`` or ``load_adapter_fusion()``.
+```
+
+To learn how training an _AdapterFusion_ layer works, check out [this Colab notebook](https://colab.research.google.com/github/Adapter-Hub/adapter-transformers/blob/master/notebooks/03_Adapter_Fusion.ipynb) from the `adapter-transformers` repo.
+
+In v1.x of `adapter-transformers`, fusing adapters was done using a nested list of adapter names, i.e. the example from above would be defined as `[["d", "e", "f"]]`.
+For backwards compatibility, you can still do this, although it is recommended to use the new syntax.
+
```

Comment:
I think at some point the add_fusion function should be mentioned and explained when to use it.

Reply:
Actually thought I already did that. Added it now, thanks :)